utils/handleToasts.ts

```diff
@@ -1,49 +1,57 @@
 import { toast } from 'react-toastify';
 
-function generateToastID() {
-  return checkIsIdOccupied(randomId());
-}
-
-function randomId() {
-  return Math.floor(Math.random() * 100000000).toString();
-}
-
-function checkIsIdOccupied(toastId: string) {
-  if (document.getElementById(toastId.toString()) !== null) {
-    return checkIsIdOccupied(randomId());
-  }
-  return toastId;
-}
-
 export function handleErrorToast(errorText: string) {
   try {
-    const generatedToastID = generateToastID();
-    if (!document) {
-      return;
-    }
-
-    toast(errorText, {
+    const toastId = toast(errorText, {
       type: 'error',
-      toastId: generatedToastID,
-      onOpen: () => {
-        setTimeout(() => {
-          document
-            .getElementById(generatedToastID)
-            ?.addEventListener('mouseenter', () => {
-              toast.update(generatedToastID, {
-                progress: 1,
+      toastId: window.crypto.randomUUID(),
+    });
+
+    const observer = new MutationObserver((mutationsList) => {
+      for (const mutation of mutationsList) {
+        mutation.addedNodes.forEach((node) => {
+          // check if node is an HTMLElement which unlock a usage of node.children
+          if (node instanceof HTMLElement) {
+            // if there are more than 1 toast, this should be working
+            if (node.id === toastId.toString()) {
+              node.addEventListener('mouseenter', () => {
+                toast.update(toastId, { progress: 1 });
               });
-            });
-          document
-            .getElementById(generatedToastID)
-            ?.addEventListener('mouseleave', () => {
-              toast.update(generatedToastID, {
-                progress: 0,
+              node.addEventListener('mouseout', () => {
+                toast.update(toastId, { progress: 0 });
               });
-            });
-        }, 1);
-      },
+              observer.disconnect();
+              return;
+            }
+
+            // the first toast seems to be always foundable using this for-loop
+            for (let x = 0; x < node.children.length; x++) {
+              const childNode = node.children[x];
+              if (childNode.id === toastId.toString()) {
+                // checking if childNode is an HTMLElement so correct event listener list will be shown
+                // in theory this might not be needed but added for now
+                if (childNode instanceof HTMLElement) {
+                  childNode.addEventListener('mouseenter', () => {
+                    toast.update(toastId, { progress: 1 });
+                  });
+                  childNode.addEventListener('mouseleave', () => {
+                    toast.update(toastId, { progress: 0 });
+                  });
+                  observer.disconnect();
+                  return;
+                }
+              }
+            }
+          }
+        });
+      }
```

Comment:
Jos Toasti on ensimmäinen, joudutaan toteuttamaan tämä. En tiedä ollenkaan onko järkevä tapa :smile: 

Eli loopataan läpi kaikki childnodet ja katsotaan onko sen alla olevan Toastin id sama kuin mitä halutaan. Tähän mennessä aina on ollut, siirrytään eteenpäin. Tässä kohtaa tarkistetaan onko `childNode` HTMLElement, jotta saadaan oikeat event listenerien vaihtoehdot

Rivi 41: Tuo `return` lienee järkevämpi vaihtaa `break`iksi, koska muuten looppaa turhaan listan loppuun

Reply:
Yritin kokeilla tuota, mutta en saanut toimimaan. Kokeilen uudelleen illemmalla, jos olinkin laittanut väärään kohtaan koodia tsekkauksen 👍

Reply:
Toimi näppärästi `document.getElementById()`. Jotain olin tehnyt väärin, kun ei aiemmin toiminut :smile: :+1: